plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VmwareVmImplementer.java

```diff
@@ -139,6 +141,9 @@ VirtualMachineTO implement(VirtualMachineProfile vm, VirtualMachineTO to, long c
                     details.put(VmDetailConstants.NIC_ADAPTER, VirtualEthernetCardType.E1000.toString());
                 }
             }
+            if(StringUtils.isEmpty(details.get(VmDetailConstants.RAM_RESERVATION))){
```

Comment:
Could cause an issue if the user does not want to reserve memory for a VM, better to leave it unset and check whether the key exists in getReservedMemoryMb

Reply:
Hi @davidjumani , thanks for testing. It strange that it not work this line in getReservedMemory should prevent this 
```java
if(!NumberUtils.DOUBLE_ZERO.toString().equals(vmSpec.getDetails().get(VmDetailConstants.RAM_RESERVATION))){
```
But the Method returns every time the minimum ram. 
```java
return (int) (vmSpec.getMinRam() / ResourceType.bytesToMiB);
```
Could it be in your test case that the reserved memory = minimum ram is.

Reply:
Nope, passing 0 ensures that no memory is reserved, but 0.0 does not. When 0.0 is passed, all the memory is reserved

Reply:
Strange, I'll take care of it. Thanks for the infos.